src/Chemistry.hpp

```diff
@@ -143,6 +146,11 @@ template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const R
 			}
 		});
 	}
+
+	int burnmin = burnstepsMF.min(0);
+	if (burnmin == 0) {
+		amrex::Abort("VODE integration was unsuccessful!");
+	}
```

Comment:
I see a potential issue here. We do not do the burn in all the cells, only in cells that have rho > rho_threshold. So, what is the value of `burnstepsMF` for cells where a burn is not done? Is it 0? If yes, then this won't work here.

Reply:
If you want to determine whether it succeeded based on the number of steps, then you should instead do `int burnmax = burnstepsMF.max(0);` and check whether `burnmax == max_steps_for_burn`.

Or you could create a boolean / integer multifab where it just stores true or false for success or failure.

Reply:
But we dont want to determine if the burn succeeded based on the number of steps. We want to determine if the burn succeeded based on the value of the boolean variable `chemstate.success`

Reply:
Ok. In that case, I would recommend following the implementation used in Castro in `Castro::react_state`:
https://github.com/AMReX-Astro/Castro/blob/5728a640ea7e5e7c47532402ff754e6a1a030fe6/Source/reactions/Castro_react.cpp#L407